neptunecontrib/monitoring/sklearn.py

```diff
@@ -0,0 +1,1155 @@
+    """Log sklearn regressor summary.
+
+    This method automatically logs all regressor parameters, pickled estimator (model),
+    test predictions as table, model performance visualizations and test metrics.
+
+    Regressor should be fitted before calling this function.
+
+    Make sure you created an experiment before you use this method: ``neptune.create_experiment()``.
+
+    Tip:
+        Check `Neptune documentation <https://docs.neptune.ai/integrations/scikit_learn.html>`_ for the full example.
+
+    Args:
+        regressor (:obj:`regressor`):
+            | Fitted sklearn regressor object
+        X_train (:obj:`ndarray`):
+            | Training data matrix
+        X_test (:obj:`ndarray`):
+            | Testing data matrix
+        y_train (:obj:`ndarray`):
+            | The regression target for training
+        y_test (:obj:`ndarray`):
+            | The regression target for testing
+        experiment (:obj:`neptune.experiments.Experiment`, optional, default is ``None``):
+            | Neptune ``Experiment`` object to control to which experiment you log the data.
+            | If ``None``, log to currently active, and most recent experiment.
+
+    Returns:
+        ``None``
+
+    Examples:
+        Log random forest regressor summary
+
+        .. code:: python3
+
+            rfr = RandomForestRegressor()
+            rfr.fit(X_train, y_train)
+
+            neptune.init('my_workspace/my_project')
+            neptune.create_experiment()
+
+            log_regressor_summary(rfr, X_train, X_test, y_train, y_test)
+    """
+    assert is_regressor(regressor), 'regressor should be sklearn regressor.'
+
+    exp = _validate_experiment(experiment)
+
+    log_estimator_params(regressor, exp)
+    log_pickled_model(regressor, exp)
+
+    y_pred = regressor.predict(X_test)
+    log_test_predictions(regressor, X_test, y_test, y_pred=y_pred, experiment=exp)
+    log_test_scores(regressor, X_test, y_test, y_pred=y_pred, experiment=exp)
+
+    # visualizations
+    log_learning_curve_chart(regressor, X_train, y_train, experiment=exp)
+    log_feature_importance_chart(regressor, X_train, y_train, experiment=exp)
+    log_residuals_chart(regressor, X_train, X_test, y_train, y_test, experiment=exp)
+    log_prediction_error_chart(regressor, X_train, X_test, y_train, y_test, experiment=exp)
+    log_cooks_distance_chart(regressor, X_train, y_train, experiment=exp)
+
+
+def log_classifier_summary(classifier, X_train, X_test, y_train, y_test, experiment=None):
+    """Log sklearn classifier summary.
+
+    This method automatically logs all classifier parameters, pickled estimator (model),
+    test predictions, predictions probabilities as table, model performance visualizations and test metrics.
+
+    Classifier should be fitted before calling this function.
+
+    Make sure you created an experiment before you use this method: ``neptune.create_experiment()``.
```

Comment:
It suggest that neptune.create_experiment is THIS method.

Reply:
thanks, changed that.